### ISSUE TITLE
CNN Image Retrieval Learns from BoW: Unsupervised Fine-Tuning with Hard Examples

### ISSUE THREAD

**Issue by chullhwan-song** [chullhwan-song/Reading-Paper#19]
https://arxiv.org/abs/1604.02426

**Comment by chullhwan-song** [comment]
### what?
* 학습셋을 만들기 너무 어렵다. 자동적 labeling 방법을 이용한 방법 제시 
   * sfm - 3d vision 한 분야 - sift같은 걸 이용하여 matching image > clustering > 3d reconstruction 
* finetune cnn for image retrieval
* hard example collection - hard negative & hard positive > more high performance
* 기본적으로 mac feature를 적용
* Siamese network training
* 이 연구는 [Kaggle Google Landmark Retrieval Challenge](https://www.kaggle.com/c/landmark-retrieval-challenge) 에서 많이 언급햇던 논문중의 하나임
   * 이유는 Google Landmark 데이터 셋 자체가 weakly labeling dataset임. > 이런 이유로..추측
* 비슷한 연구가 많이 있는데, mac & metric learning 조합으로 많이 사용/논문 존재
    * 참고로 여기서는 R-MAC이 아니고 original mac임.
    * DIR 논문의 originality가 여기에 있는듯해 보임.

### image representation
* Maximum Activations of Convolutions (MAC)
![image](https://user-images.githubusercontent.com/40360823/42740081-032862a4-88dd-11e8-9837-29aaf113e8e2.png)
    * input : W x H x K
    * output : K
    * f : non-negative > ReLU
    * next >  l2 normalized
    * similarity > dot product 
        * fig.1 은 image patche에 대한 similarity 
        * fine-tuning
![image](https://user-images.githubusercontent.com/40360823/42740083-0c46d0e6-88dd-11e8-9e15-ab36e0a48284.png)

### Network and siamese learning
* metric learning - Siamese network training
* AlexNet & VGG.
* In this paper, we focus on **re-training (i.e. fine-tuning)** state-of-the-art CNNs for classification.
* for training
   * image pairs (i, j) and labels Y(i,  j) = { 0, 1} declaring whether a pair is non-matching (label 0) or matching (label 1).
   * contrastive loss
![image](https://user-images.githubusercontent.com/40360823/42740086-16f9c282-88dd-11e8-85e0-6104e4dae2bd.png)
       * ![image](https://user-images.githubusercontent.com/40360823/42740092-20952eb2-88dd-11e8-827f-ae46481b5e70.png) : when non-matching pairs(negative pair) have large enough distance in order not to be
taken into account in the loss.
    * Negative images
       * Negative examples are selected from clusters different than the cluster of the query image.
       * choose hard negatives
            * metric learning에서는 이게 핵심인듯~
            * non-matching images with the most similar descriptor. > 그러니까 다른 클래스의 pair를 선택하고 서로 매우 유사한 feature를 같는 것을 선택해야한다.
            * k-nearest neighbors

### Whitening and dimensionality reduction
* PCA > l2 normalized

### Performance
![image](https://user-images.githubusercontent.com/40360823/42740095-31d04a9a-88dd-11e8-9769-2e3a13cabed9.png)
![image](https://user-images.githubusercontent.com/40360823/42740096-33a0e050-88dd-11e8-8204-2ce5a071875d.png)
![image](https://user-images.githubusercontent.com/40360823/42740098-364ccdc8-88dd-11e8-988a-4aa17abce25a.png)
